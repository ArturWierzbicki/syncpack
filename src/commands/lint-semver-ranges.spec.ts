import 'expect-more-jest';
import * as mock from '../../test/mock';
import { DEFAULT_CONFIG } from '../constants';
import * as api from './lint-semver-ranges';

describe('lint-semver-ranges', () => {
  let lintSemverRanges: typeof api.lintSemverRanges;
  let log: jest.Mock;

  afterEach(() => {
    jest.restoreAllMocks();
  });

  beforeEach(() => {
    jest.mock('./lib/log', () => ({ log: jest.fn() }));
    lintSemverRanges = require('./lint-semver-ranges').lintSemverRanges;
    log = require('./lib/log').log;
  });

  it('outputs all dependencies with incorrect versions', () => {
    const wrappers = [
      mock.wrapper('a', ['foo@0.1.0'], [], [], { name: 'pkg1' }),
      mock.wrapper('b', ['foo@0.2.0', 'bar@^0.2.0', 'baz@~0.3.0'], [], [], { name: 'pkg2' }),
    ];
    lintSemverRanges(wrappers, { ...DEFAULT_CONFIG, dev: false, peer: false, prod: true });
    expect(log.mock.calls).toMatchSnapshot();
  });
});
