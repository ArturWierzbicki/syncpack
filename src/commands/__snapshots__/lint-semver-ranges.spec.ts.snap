// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint-semver-ranges outputs all dependencies with incorrect versions 1`] = `
Array [
  Array [
    "[31mâœ• bar[39m ^0.2.0 [2min dependencies of pkg2[22m",
  ],
  Array [
    "[31mâœ• baz[39m ~0.3.0 [2min dependencies of pkg2[22m",
  ],
]
`;
